### PR TITLE
onnxml and SKL Label Encoder

### PR DESCRIPTION
**Pull request by ksaur**
This adds LabelEncoder for SKL (strings and ints) and ONNX (ints).

### CODE REVIEW

**Comment by ksaur** [comment]
closes #160

**Comment by ksaur** [comment]
Ok i think we have a good hybrid to be prepared for torch==1.8.  PTAL @interesaaat !

File: .github/workflows/pythonapp.yml [hunk 1]
```diff
@@ -47,9 +47,9 @@ jobs:
     - name: Install pytorch 1.5.1+cpu if python 3.5 (not mac)
       if: ${{ matrix.python-version == '3.5' && matrix.os != 'macos-10.14' }}
       run: pip install torch==1.5.1+cpu -f https://download.pytorch.org/whl/torch_stable.html
-    - name: Install pytorch 1.7.0+cpu if python > 3.5 (not mac)
+    - name: Install pytorch nightly if python > 3.5 (not mac)
       if:  ${{ matrix.python-version != '3.5' && matrix.os != 'macos-10.14' }}
-      run: pip install torch==1.7.0+cpu -f https://download.pytorch.org/whl/torch_stable.html
+      run: pip install --pre torch -f https://download.pytorch.org/whl/nightly/cpu/torch_nightly.html
```
Comment (interesaaat) [inline .github/workflows/pythonapp.yml]: I am not sure we want to run our cicd on their nightly built. Since it's not stable we may get intro troubles.

File: tests/test_onnxml_label_encoder_converter.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        ]
+        model.fit(data)
+
+        # max word length is the smallest number which is divisible by 4 and larger than or equal to the length of any word
+        max_word_length = 12
+        view_args = [-1, max_word_length // 4]
+        str_dtype = "|S" + str(max_word_length)
+        np_arr = np.array(data, dtype=str_dtype).view(np.int32)
+
+        # pytorch_input = torch.from_numpy(np_arr).view(*view_args)
```
Comment (interesaaat) [inline tests/test_onnxml_label_encoder_converter.py:79]: This can be removed

**Review by interesaaat — APPROVED** [review]
Just have 2 small comments. Do you want to fix them, or I can.

File: .github/workflows/pythonapp.yml (hunk 1 above)
Comment (ksaur) [reply .github/workflows/pythonapp.yml]: ok, since we need 1.8 but it's not in pypi yet, do you suggest we pin it to a [specific version](https://download.pytorch.org/whl/nightly/cu110/torch_nightly.html) of 1.8? If so, which?

Comment (interesaaat) [reply .github/workflows/pythonapp.yml]: We need 1.8 only for exporting to onnx, pytorch should still work (TVM, I don't know). So I think it is safer to leave 1.7 in the pipeline and move to 1.8 once it's stable. When the onnx export is used, we can either leave as it is (so it will say `nonzero` is not currently supported), or we can add a specific exception, like label encoder not supported until pytorch 1.8.

Comment (ksaur) [reply .github/workflows/pythonapp.yml]: k sounds good
